src/main/java/com/o19s/es/ltr/query/RankerQuery.java

```diff
@@ -0,0 +1,320 @@
+            }
+            float modelScore = ranker.score(d);
+            return Explanation.match(modelScore, " LtrModel: " + ranker.name() + " using features:", subs);
+        }
+
+        @Override
+        public float getValueForNormalization() throws IOException {
+            // XXX: may produce inconsistent scores if featured queries
+            // are run individually and a Similarity that implements queryNorm is used.
+            // queryNorm will disappear in lucene 7.
+            // Should we hack something or warn if this query is being used with ClassicSimilarity?
+            float sum = 0.0f;
+            for (Weight w : weights) {
+                sum += w.getValueForNormalization();
+            }
+            return sum ;
+        }
+
+        @Override
+        public void normalize(float norm, float boost) {
+            for(Weight w : weights) {
+                w.normalize(norm, boost);
+            }
+        }
+
+        @Override
+        public RankerScorer scorer(LeafReaderContext context) throws IOException {
+            ArrayList<Scorer> scorers = new ArrayList<>(weights.length);
+            List<DocIdSetIterator> subIterators = new ArrayList<>(weights.length);
+            for (int i = 0; i < weights.length; i++) {
+                Scorer scorer = weights[i].scorer(context);
+                if (scorer == null) {
+                    scorer = new NoopScorer(this, context.reader().maxDoc());
+                }
+                scorers.add(scorer);
+                subIterators.add(scorer.iterator());
+            }
+            DocIdSetIterator rankerIterator = new NaiveDisjunctionDISI(DocIdSetIterator.all(context.reader().maxDoc()), subIterators);
+            return new RankerScorer(scorers, rankerIterator);
+        }
+
+        class RankerScorer extends Scorer {
+            /**
+             * NOTE: Switch to ChildScorer and {@link #getChildren()} if it appears
+             * to be useful for logging
+             */
+            private final ArrayList<Scorer> scorers;
+            private final DocIdSetIterator iterator;
+            private final LtrRanker.DataPoint dataPoint;
+
+            RankerScorer(ArrayList<Scorer> scorers, DocIdSetIterator iterator) {
+                super(RankerWeight.this);
+                this.scorers = scorers;
+                this.iterator = iterator;
+                dataPoint = ranker.newDataPoint();
+            }
+
+            @Override
+            public int docID() {
+                return iterator.docID();
+            }
+
+            @Override
+            public float score() throws IOException {
+                for (int i = 0; i < scorers.size(); i++) {
+                    Scorer scorer = scorers.get(i);
+                    if (scorer.docID() == docID()) {
+                        dataPoint.setFeatureScore(i, scorer.score());
+                    } else {
+                        dataPoint.setFeatureScore(i, 0);
```

Comment:
Sure, then I'll probably defer this to the feature vector and implement a reset() method and remove the else part or a `reuse` param as seen sometimes in lucene...